spigot/src/main/java/fr/freebuild/playerjoingroup/spigot/MessagesManager.java

```diff
@@ -174,10 +174,10 @@ public void send(byte[] msg) throws IOException {
     }
 
     private void onPlayerLeave(Packet packet) {
-        String playerName = packet.getParams().get("PLAYER_NAME");
-        String message = Utils.getConfigString("QuitMessage");
-        message = Utils.format(message, FormatParam.PLAYER, playerName);
-        getServer().broadcastMessage(message);
+        if (canDisplayMessage(packet, "essentials.silentquit"))  {
+            String message = Utils.getPlayerLeaveMessage(packet.getParams().get("PLAYER_NAME"));
+            getServer().broadcastMessage(message);
+        }
```

Comment:
Doesn't work because the player already leave. Find a way to check permission on offline player or move the check elsewhere

Reply:
If we can apply the fix to this problem to the [canDisplayMessage](https://github.com/G-Lauz/PlayerJoinGroup/pull/4/files#diff-845a76d29fb8661d33886b38cd4cc676a1a6568fc62c82da3bf027738b027413R220-R225) method and make it work in any case this could be clean!

Reply:
`LEAVE_SERVER_GROUP` event are trigger and sent by `bungee.listener.PlayerDisconnectListener` and `bungee.listener.PlayerSwitchListener` if needed.

Another naïve solution that I see is we can add a flag (ParamKey) to the packet according to the player status and only send the message to those having the permission (something like that).

In both case I think it's not ideal and sometime ugly. Maybe we can find something better.